### DISCUSSION TITLE
[litehtml] No DLL on Windows

### DISCUSSION THREAD

**Discussion by rioki** [microsoft/vcpkg#29612] [Q&A] [answered]
I am currently working on a port for [litehtml](https://github.com/litehtml/litehtml). They have a CMakeLists.txt, so integration was mostly flawless; except on Windows. 

It appears that on windows, building with Visual Studio is the preferred mode and then it is a static library. I patched the CMakeLists.txt to generate a static library. This feels like it goes against the grain for `x64-windows` and `x86-windows`. Should I rather extend the source code to create proper C++ DLLs?

(I am considering submitting a DLL patch upstream, but that won't help me for now.)

**Comment by dg0yt** [comment, marked as answer]
You may start with a static lib in dynamic windows triplets by this portfile code:
~~~cmake
if (VCPKG_TARGET_IS_WINDOWS)
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
endif()
~~~
It is a frequent pattern, e.g. used by port minizip.

**Reply by dg0yt** [reply]
Note that this also changes the default library type to `STATIC` so you may not need a patch.

**Reply by rioki** [reply]
This sounds like a sound solution.
